### ISSUE TITLE
Extracting loss over epochs

### ISSUE THREAD

**Issue by axelurbiz** [statsmaths/kerasR#7]
Hello Taylor et al.,

Is there a way to extract the loss and metrics from a `keras_fit` object or alternative ways of doing so?

Here's what I'm looking to replicate in R from Python:

```python
# Visualize training history
from keras.models import Sequential
from keras.layers import Dense
from keras.utils import np_utils
import matplotlib.pyplot as plt
from sklearn import datasets
import numpy

# fix random seed for reproducibility
seed = 7
numpy.random.seed(seed)

# import some data to play with
iris = datasets.load_iris()
X = iris.data[:, :4]  
Y = iris.target
Y = np_utils.to_categorical(Y)

# create model
model = Sequential()
model.add(Dense(4, input_dim = 4, kernel_initializer = 'normal', activation='relu'))
model.add(Dense(3, kernel_initializer='normal', activation='softmax'))

# Compile model
model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

# Fit the model
history = model.fit(X, Y, validation_split=0.33, epochs=150, batch_size=5, verbose=0)

# list all data in history
print(history.history.keys())

# summarize history for accuracy
plt.plot(history.history['acc'])
plt.plot(history.history['val_acc'])
plt.title('model accuracy')
plt.ylabel('accuracy')
plt.xlabel('epoch')
plt.legend(['train', 'test'], loc='upper left')
plt.show()

# summarize history for loss
plt.plot(history.history['loss'])
plt.plot(history.history['val_loss'])
plt.title('model loss')
plt.ylabel('loss')
plt.xlabel('epoch')
plt.legend(['train', 'test'], loc='upper left')
plt.show()
```

Here's as far as I could get from the analogous R script. I cannot seem to extract the loss/accuracy from either the `model` object or `keras_fit` results.

```r
library(kerasR)

X <- as.matrix(iris[, 1:4])
Y <- to_categorical(as.numeric(iris$Species))[, 2:4]

model <- Sequential()
model$add(Dense(units = 4, input_shape = 4, kernel_initializer = "normal"))
model$add(Activation("relu"))
model$add(Dense(units = 3, kernel_initializer = "normal"))
model$add(Activation("softmax"))
keras_compile(model, loss = "categorical_crossentropy", optimizer = Adam(), metrics = "categorical_accuracy")
history <- keras_fit(model, X, Y, batch_size = 5, epochs = 150, verbose = 0,  validation_split = 0.33)
```
> history
[1] 0

Any hint would be appreciated. 

Thanks,
A.

**Comment by statsmaths** [comment]
Sorry, missed this issue earlier. The `keras_fit` function literally fits the weights in the model. By default it returns `0` if there are no errors in the model. If you want to see all of the pure-python functions available for extracting information from the model, type `model$` and hit tab to see them printed to the screen. Alternatively, see the classes `CSVLogger` or `ModelCheckpoint` for callbacks to directly attach the fitting process to watch the output after each epoch.
